doc/coq.rst

```diff
@@ -0,0 +1,379 @@
+.. _coq:
+
+***
+Coq
+***
+
+.. contents:: Table of Contents
+    :depth: 3
+
+Introduction
+------------
+
+Dune is able to build Coq theories and plugins with support for source file
+preprocessing, and extraction.
```

Comment:
Maybe add `.mlg file preprocessing`, otherwise the sentence seems too vague to me.